### PR TITLE
doc: release: 3.5: Add Ethernet and Wi-Fi info

### PR DESCRIPTION
**Pull request by jukkar**
Add changes to Ethernet and Wi-Fi drivers for v3.5. Also add information about generic Ethernet and Wi-Fi changes.

### CODE REVIEW

File: doc/releases/release-notes-3.5.rst [hunk 1]
```diff
@@ -224,6 +224,29 @@ Drivers and Sensors
...
+  * Fixed stm32 to generate more unique MAC address by using device id as a base for the MAC.
+  * Fixed mcux to increase the PTP timestamp accuracy from 20us to 200ns.
+  * Fixed Ethernet max header size when using VLAN.
+  * Fixed mimxrt1050_evk gPTP Grand Master endpoint sync accuracy increase from 3.5ms to 580ns.
+  * Fixed eth_mcux to not use deprecated SDK functions.
+  * Removed the `mdio` DT property. Please use DT_INST_BUS() in the driver instead.
+  * Reworked the device node hierarchy in smsc91x.
+  * Renamed the phy-dev property with phy-handle to match the Linux ethernet-controller binding
+    and move it up to ethernet.yaml so that it can be used by other drivers.
+  * Updated Ethernet PHY to use `reg` property in DT bindings.
```
Comment (kartben) [inline doc/releases/release-notes-3.5.rst:246]:
```suggestion
  * Updated Ethernet PHY to use ``reg`` property in DT bindings.
```

File: doc/releases/release-notes-3.5.rst [hunk 2]
```diff
@@ -224,6 +224,29 @@ Drivers and Sensors
...
+  * Enabled the correct IRQ for multi-instance in eth_e1000.
+  * Fixed eth_e1000 to use IRQ "flags" if IRQ "sense" is not present in DT.
+  * Fixed Ethernet startup when fixed-link in GMAC.
+  * Fixed eth_smsc91x to select bank 3 when writing to MII register.
+  * Fixed stm32 to generate more unique MAC address by using device id as a base for the MAC.
+  * Fixed mcux to increase the PTP timestamp accuracy from 20us to 200ns.
+  * Fixed Ethernet max header size when using VLAN.
+  * Fixed mimxrt1050_evk gPTP Grand Master endpoint sync accuracy increase from 3.5ms to 580ns.
+  * Fixed eth_mcux to not use deprecated SDK functions.
+  * Removed the `mdio` DT property. Please use DT_INST_BUS() in the driver instead.
```
Comment (kartben) [inline doc/releases/release-notes-3.5.rst:242]:
```suggestion
  * Removed the ``mdio`` DT property. Please use :c:macro:`DT_INST_BUS()` in the driver instead.
```

File: doc/releases/release-notes-3.5.rst [hunk 3]
```diff
@@ -224,6 +224,29 @@ Drivers and Sensors
...
+  * Reworked the device node hierarchy in smsc91x.
+  * Renamed the phy-dev property with phy-handle to match the Linux ethernet-controller binding
+    and move it up to ethernet.yaml so that it can be used by other drivers.
+  * Updated Ethernet PHY to use `reg` property in DT bindings.
+  * Updated driver DT bindings to use `ethernet-phy` devicetree node name consistently.
```
Comment (kartben) [inline doc/releases/release-notes-3.5.rst:247]:
```suggestion
  * Updated driver DT bindings to use ``ethernet-phy`` devicetree node name consistently.
```

File: doc/releases/release-notes-3.5.rst [hunk 4]
```diff
@@ -361,6 +384,19 @@ Drivers and Sensors
 
 * WiFi
 
+  * Increased esp32 default network (TCP workq, RX and mgmt event) stack sizes to 2048 bytes.
+  * Added support for esp32c3_luatos_core board.
+  * Added support for esp32s3_luatos_core board.
+  * Added support for xiao_esp32s3 board.
```
Comment (kartben) [inline doc/releases/release-notes-3.5.rst:390]: mmmm should just be listed in the Xtensa boards?

File: doc/releases/release-notes-3.5.rst [hunk 5]
```diff
@@ -388,6 +436,17 @@ Networking
...
+  * Added support for detecting whether the Wi-Fi driver is offloaded or native one.
+  * Added support for generic network manager API.
+  * Added support for W-Fi mode setting and selection.
+  * Added user input validation for SSID and PSK in Wi-Fi shell.
+  * Added scan extension for specifying channels.
+  * Added scan extension for limiting scan results.
+  * Added scan extension for filtering SSIDs.
+  * Added scan extension for passive channel dwell times.
+  * Added scan extension for active channel dwell times.
+  * Added scan extension support for frequency bands.
```
Comment (kartben) [inline doc/releases/release-notes-3.5.rst:449]: Group under same item?

**Review by kartben — CHANGES_REQUESTED** [review]
Thanks! A few comments added.

Also, I am not sure I am a big fan of listing virtually _all_ changes, so maybe there are some that can be just skipped, or lumped into the same bullet item with some rewording to not just paraphrase the original PR? Things like "Fixed Announce message byte order" are not necessarily helpful IMO (especially if it's not linking to the PR/issue for folks to get all the background).

File: doc/releases/release-notes-3.5.rst (hunk 4 above)
Comment (jukkar) [reply doc/releases/release-notes-3.5.rst:390]: These are related to wifi support in those boards. I can certainly remove all of them from this section.

File: doc/releases/release-notes-3.5.rst (hunk 5 above)
Comment (jukkar) [reply doc/releases/release-notes-3.5.rst:449]: Sure

**Comment by jukkar** [comment]
* Removed some minor changes
* Grouped couple of items together

**Review by jhedberg — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
`/runner/_work/zephyr/zephyr/doc/_build/src/releases/release-notes-3.5.rst:436: ERROR: Unexpected indentation.`

File: doc/releases/release-notes-3.5.rst [hunk 6]
```diff
@@ -388,6 +422,12 @@ Networking
 * Wi-Fi
   * Added Passive scan support.
   * The Wi-Fi scan API updated with Wi-Fi scan parameter to allow scan mode selection.
+  * Updated TWT handling.
+  * Added support for generic network manager API.
+  * Added support for W-Fi mode setting and selection.
```
Comment (kartben) [inline doc/releases/release-notes-3.5.rst:427]: apologies if I missed this one earlier. I will approve nevertheless as we can also address pending typos one last time before the release, I guess.

```suggestion
  * Added support for Wi-Fi mode setting and selection.
```

**Review by kartben — DISMISSED** [review]

**Review by rlubos — DISMISSED** [review]

**Comment by jukkar** [comment]
Fixed the typo and re-submitted

**Review by kartben — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**